Description:  As of glibc 2.9, struct hostent no longer defines h_addr, which was a backward-compatability
 alias for h_addr_list[0]
Bug-Debian: 491798.
Last-Updated: 2009-03-27
Author: Alastair McKinstry <mckinstry@debian.org>
Forwarded: no

diff -ruN slang-2.2.2-old/modules/socket-module.c slang-2.2.2/modules/socket-module.c
--- slang-2.2.2-old/modules/socket-module.c	2008-02-25 05:28:07.000000000 +0000
+++ slang-2.2.2/modules/socket-module.c	2009-03-28 15:01:24.000000000 +0000
@@ -445,7 +445,7 @@
 #ifndef h_addr
    /* Older interface.  There is only one address, so fake a list */
    h_addr_list = fake_h_addr_list;
-   h_addr_list [0] = hp->h_addr;
+   h_addr_list [0] = hp->h_addr_list[0];
    h_addr_list [1] = NULL;
 #else
    h_addr_list = hp->h_addr_list;
